final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,183 @@
+import argparse
+from typing import Union
+
+import feedparser
+import logging
+import json
+import html
+import sys
+import datetime
+import os
+
+__VERSION__ = '3.5'
+ARGS = None
+LIMIT = 3
+LINKS = []
+PRINT_SETTING = {
+    'feed': [
+        'title'
+    ],
+    'entries': [
+        'title',
+        'published',
+        'link',
+        'summary'
+    ]
+}
+
+
+def get_object_feed(url: str) -> Union[str, feedparser.FeedParserDict]:
+    try:
+        data = feedparser.parse(url)
+        if data.status == 200:
+            if data.bozo:
+                return f'ERROR: There is no rss feed at this url: {url}'
+            else:
+                return data
+        else:
+            return f'HTTP Status Code {data.status}'
+    except Exception as exc:
+        return f'ERROR: {exc}'
+
+
+def to_json(data) -> json:
+    """convert data to JSON format"""
+    result = {}
+    if isinstance(data, feedparser.FeedParserDict):
+        if 'feed' in PRINT_SETTING.keys():
+            for feed_element in PRINT_SETTING['feed']:
+                result[feed_element] = text_processing(data['feed'][feed_element])
+        if 'entries' in PRINT_SETTING.keys():
+            result['items'] = []
+            for index_news in range(LIMIT):
+                temp = {}
+                for items_element in PRINT_SETTING['entries']:
+                    temp[items_element] = text_processing(data['entries'][index_news][items_element])
+                result['items'].append(temp)
+        result['links'] = LINKS
+    result = json.dumps(result)
+    return result
+
+
+def get_img(input_string: str) -> [str, str]:
+    """from string type '<img src="link" alt="something">text' returns ['link', 'something']"""
+    input_string = input_string[input_string.find('<img'):]
+    link = input_string[input_string.find('src="') + 5:]
+    str_img = input_string[input_string.find('alt="') + 5:]
+    return [link[:link.find('"')], str_img[:str_img.find('"')]]
+
+
+def text_processing(string):
+    """processes a string for output"""
+    global LINKS
+    image = ''
+    if '<' not in string:
+        return html.unescape(string)
+    if 'img' in string:
+        image_link_and_alt_text = get_img(string)
+        LINKS.append(image_link_and_alt_text[0])
+        image = '[image ' + str(len(LINKS)) + ': ' + image_link_and_alt_text[1] + '][' + str(len(LINKS)) + '] '
+    string = string[string.find('<') + 1:]
+    while string.find('<') - string.find('>') < 2:
```

Comment:
This line can under some circumstances freeze your program. See point 6 of my feedback.